client/jetpack-connect/authorize.js

```diff
@@ -888,7 +888,7 @@ export class JetpackAuthorize extends Component {
 		const productSlug = searchParams.get( 'productSlug' );
 		const siteSlug = searchParams.get( 'fromSiteSlug' );
 		const product = getProductFromSlug( productSlug );
-		const productName = getJetpackProductDisplayName( product );
+		const productName = getJetpackProductOrPlanDisplayName( product );
```

Comment:
After introducing that function, `productName` can implicitly be `undefined`. That might lead to visual bugs in the text displayed to user.

We should either clean up the types (if we can be certain product name will be defined) or guard that edge case.